platform-sdk/docs/proposals/DP-0x/0X - Tss block signing - Cryptography APIs.md

```diff
@@ -0,0 +1,498 @@
+
+###### 2. Validation of TssMessage
+Each participant will validate the received message against the commitment and the zk-SNARKs proof. Invalid messages will be discarded.
+The validation is produced over the content of the message and does not include the sender's identity which is assumed to be provided by the external channel.
+
+###### 3. Processing of TssMessage
+Given Participant's persistent EC PrivateKey, `n` number of validated messages (n=threshold), and the list of all validated `TssMessage`s,
+each participant will decrypt all `Mₛ` to generate an aggregated value `xₛ` that will become a  `ProprietaryShare(sidₛ, xₛ)` for each `ShareId`: `sidₛ` owned by the participant.
+
+
+![img_8.png](img_8.png)
+
+At this point, the participant executing the scheme is able to start signing.
+Synchronization between all participants will be needed in order to determine if all of them have received and processed enough messages and determined each owns ProprietaryShares,
+but that is not covered by this library and should be done outside.
+
+#####  Sign Stage
+
+###### 1. Sign
+Using each `ProprietaryShare` owned by the participant, a message can be signed producing a `TssSignature`
+Using a channel established outside the scope of the scheme and a message also created outside the scope of the library, each participant will broadcast a `TssSignature` per `ProprietaryShare` to the other participants.
+
+###### 2. Aggregate Signature
+Once the participant has received a number of messages `n` (n=threshold), validate them (?) and process them in sequence (?).
+Extract the signature and aggregate the signature for all messages.
+(?)Synchronization between all participants will be needed in order to determine if all of them have received and processed enough messages and determined the AggregatedSignature,
+
+###### 3. Persist
+The aggregate signature can be persisted.
+
+#####  Rekey Stage
+The rekeying process is similar to the bootstrap process 
+
+.g: In a scheme distributing 10 shares over 4 participants
+
+Assuming Current Participant: (P₁;  P₁_EC_PrivateKey)
+
+Given
+
+The current participants' directory:
+```
+P   # shares
+-----------------------------
+P₁  3        P₁_EC_PublicKey
+P₂  2        P₂_EC_PublicKey
+P₃  1        P₃_EC_PublicKey
+P₄  2        P₄_EC_PublicKey
+P₅  1        P₅_EC_PublicKey
+P₆  1        P₆_EC_PublicKey
+```
+A threshold:
+`5`
+
+An old ownership map: `ShareId`->`Participant`:
+```
+sid₁	sid₂	sid₃	sid₄	sid₅	sid₆	sid₇	sid₈	sid₉	sid₁₀
+P₁  	P₁  	P₁  	P₁  	P₁  	P₂  	P₂  	P₃  	P₄  	P₄  
+```
+
+And a new ownership map: `ShareId`->`Participant`:
+```
+sid₁	sid₂	sid₃	sid₄	sid₅	sid₆	sid₇	sid₈	sid₉	sid₁₀
+P₁  	P₁  	P₁  	P₂  	P₂  	P₃  	P₄  	P₄  	P₅  	P₆  
+```
+
+but instead, each participant will generate a `TssMessage` out of each `ProprietaryShare` owned by the participant.
+Once finished, the list of ProprietariesShares will be updated 
+
+[TODO]
+
+
+### Architecture
+
+![img_6.png](img_6.png)
```

Comment:
Is the green box indicating a module? What do the various colors mean? What do the arrow means? I think a legend would help.